src/node/dns.ts

```diff
@@ -1,30 +1,70 @@
-import { default as dnsUtil } from 'node-internal:dns';
 import * as errorCodes from 'node-internal:internal_dns_constants';
+import {
+  getServers,
+  lookup,
+  lookupService,
+  resolve,
+  resolve4,
+  resolve6,
+  resolveAny,
+  resolveCname,
+  resolveCaa,
+  resolveMx,
+  resolveNaptr,
+  resolveNs,
+  resolvePtr,
+  resolveSoa,
+  resolveSrv,
+  setDefaultResultOrder,
+  getDefaultResultOrder,
+  setServers,
+} from 'node-internal:internal_dns';
+import * as dns from 'node-internal:internal_dns';
 
 export * from 'node-internal:internal_dns_constants';
+export {
+  getServers,
+  lookup,
+  lookupService,
+  resolve,
+  resolve4,
+  resolve6,
+  resolveAny,
+  resolveCname,
+  resolveCaa,
+  resolveMx,
+  resolveNaptr,
+  resolveNs,
+  resolvePtr,
+  resolveSoa,
+  resolveSrv,
+  setDefaultResultOrder,
+  getDefaultResultOrder,
+  setServers,
+} from 'node-internal:internal_dns';
+
+function toCallback<A = unknown, ReturnValue = unknown>(
+  fn: (arg: A) => Promise<ReturnValue>
+): (
+  input: A,
+  callback: (error: Error | null, value?: ReturnValue) => void
+) => void {
+  return (
+    input: A,
+    callback: (error: Error | null, value?: ReturnValue) => void
+  ) => {
+    fn(input)
+      .then((value) => {
+        callback(null, value);
+      })
+      .catch((error: unknown) => {
+        callback(error as Error);
+      });
+  };
+}
 
-export const getServers = dnsUtil.getServers.bind(dnsUtil);
-export const lookup = dnsUtil.lookup.bind(dnsUtil);
-export const lookupService = dnsUtil.lookupService.bind(dnsUtil);
-export const resolve = dnsUtil.resolve.bind(dnsUtil);
-export const resolve4 = dnsUtil.resolve4.bind(dnsUtil);
-export const resolve6 = dnsUtil.resolve6.bind(dnsUtil);
-export const resolveAny = dnsUtil.resolveAny.bind(dnsUtil);
-export const resolveCname = dnsUtil.resolveCname.bind(dnsUtil);
-export const resolveCaa = dnsUtil.resolveCaa.bind(dnsUtil);
-export const resolveMx = dnsUtil.resolveMx.bind(dnsUtil);
-export const resolveNaptr = dnsUtil.resolveNaptr.bind(dnsUtil);
-export const resolveNs = dnsUtil.resolveNs.bind(dnsUtil);
-export const resolvePtr = dnsUtil.resolvePtr.bind(dnsUtil);
-export const resolveSoa = dnsUtil.resolveSoa.bind(dnsUtil);
-export const resolveSrv = dnsUtil.resolveSrv.bind(dnsUtil);
-export const resolveTxt = dnsUtil.resolveTxt.bind(dnsUtil);
-export const reverse = dnsUtil.reverse.bind(dnsUtil);
-export const setDefaultResultOrder =
-  dnsUtil.setDefaultResultOrder.bind(dnsUtil);
-export const getDefaultResultOrder =
-  dnsUtil.getDefaultResultOrder.bind(dnsUtil);
-export const setServers = dnsUtil.setServers.bind(dnsUtil);
+export const reverse = toCallback(dns.reverse.bind(dns));
+export const resolveTxt = toCallback(dns.resolveTxt.bind(dns));
```

Comment:
Since you're going from promises to callbacks here, how are you handling synchronously throwing on argument validation? By default the JSG wrapper should cause synchronously thrown errors at the c++ level to be returned into promise rejections. Are you certain this will do the right thing for the errors the Node.js API expects to be thrown synchronously?

Reply:
I'll remove `async` from functions and just return a promise from the function, which would throw validation errors synchronously but the function will return Promise